internal/clientconn/conninfo/conn_info.go

```diff
@@ -37,6 +40,35 @@ type ConnInfo struct {
 	rw       sync.RWMutex
 	username string
 	password string
+
+	clientMetadata ClientMetadata
+}
+
+type ClientMetadata struct {
+	K           []string    `json:"$k"`
```

Comment:
Those are not really JSON values, that's just a logging representation of the `*types.Document`. On the wire, it is encoded in BSON.

I think we should be good just by keeping track of if metadata was already received:
```go
metadataReceived bool
```